### PR TITLE
added dictionary type

### PR DESCRIPTION
**Pull request by iamchiwon**
for [String:Any] type dictionary.

### CODE REVIEW

File: Sources/RxCodable/ObservableType+RxCodable.swift [hunk 1]
```diff
@@ -17,3 +17,11 @@ public extension ObservableType where E == String {
       .map(type, using: decoder)
   }
 }
+
+public extension ObservableType where E == [String: Any] {
+  public func map<T>(_ type: T.Type, using decoder: JSONDecoder? = nil) -> Observable<T> where T: Decodable {
+    return self
+      .map { dict in try JSONSerialization.data(withJSONObject: dict, options: .prettyPrinted) }
```
Comment (devxoul) [inline Sources/RxCodable/ObservableType+RxCodable.swift]: Why do we need `prettyPrinted` option?

File: Tests/RxCodableTests/RxCodableTests.swift [hunk 2]
```diff
@@ -76,4 +76,15 @@ class RxCodableTests: XCTestCase {
     XCTAssertEqual(try Single.just(jsonString).map([User].self).toBlocking().first()!, users)
     XCTAssertEqual(try Maybe.just(jsonString).map([User].self).toBlocking().first()!, users)
   }
+  
+  func testMapCodableFromDictionary() {
+    let dictionary:[String:Any] = [
+      "id":123,
+      "name":"iamchiwon",
```
Comment (devxoul) [inline Tests/RxCodableTests/RxCodableTests.swift]: Can you please add a whitespace after the colon(:)?

**Review by devxoul — CHANGES_REQUESTED** [review]

File: Sources/RxCodable/ObservableType+RxCodable.swift (hunk 1 above)
Comment (iamchiwon) [reply Sources/RxCodable/ObservableType+RxCodable.swift]: yes. it is not necessary

File: Tests/RxCodableTests/RxCodableTests.swift (hunk 2 above)
Comment (iamchiwon) [reply Tests/RxCodableTests/RxCodableTests.swift]: ok. 
it'll be like below

"id": 123,
"name": "iamchiwon",

**Comment by iamchiwon** [comment]
I have added some functions and requesting together

**Pull request closed without merging**